src/Error/Error.php

```diff
@@ -47,28 +47,43 @@
     private $filePath;
 
     /**
-     * @param int    $type
-     * @param string $filePath
+     * @var \Exception|\Error|\Throwable|null
      */
-    public function __construct($type, $filePath)
+    private $source;
+
+    /**
+     * @param int                               $type
+     * @param string                            $filePath
+     * @param \Exception|\Error|\Throwable|null $source
```

Comment:
```
\Throwable|null
```
is fine by me also as we would've likely more IDE support. Agreed we don't write code for IDE's, but this feels like a good compromise to me (and it would be more inline what we've now in the project).